From 28cf779978984cf9e3a5752e38687c1350aa1898 Mon Sep 17 00:00:00 2001
From: Wang Hui <Hui.Wang@windriver.com>
Date: Mon, 12 Nov 2012 15:58:52 +0800
Subject: [PATCH 25/50] armadaxp: nor flash: modify spi flash according to new mtd interface

The original spi flash driver is picked from kernel-3.2, in
kernel-3.4, the mtd interface has some change, so we need to modify
the driver.

Signed-off-by: Wang Hui <Hui.Wang@windriver.com>
---
 .../arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c |   20 ++++++++++----------
 drivers/mtd/maps/Makefile                          |    5 +++++
 2 files changed, 15 insertions(+), 10 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c
index b4da128..1fe6d59 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_mtd/sflash.c
@@ -161,16 +161,16 @@ static struct mtd_info *sflash_probe(struct map_info *map)
 	mtd->size = sflash->sectorSize * sflash->sectorNumber;
 	mtd->priv = map; /*sflash;*/
 	mtd->type = MTD_NORFLASH;
-	mtd->erase = sflash_erase;
-	mtd->read = sflash_read;
-	mtd->write = sflash_write;
-	mtd->sync = sflash_sync;
-	mtd->suspend = sflash_suspend;
-	mtd->resume = sflash_resume;	
-	mtd->lock = sflash_lock;
-	mtd->unlock = sflash_unlock;
-	mtd->block_isbad = sflash_block_isbad;
-	mtd->block_markbad = sflash_block_markbad;	
+	mtd->_erase = sflash_erase;
+	mtd->_read = sflash_read;
+	mtd->_write = sflash_write;
+	mtd->_sync = sflash_sync;
+	mtd->_suspend = sflash_suspend;
+	mtd->_resume = sflash_resume;	
+	mtd->_lock = sflash_lock;
+	mtd->_unlock = sflash_unlock;
+	mtd->_block_isbad = sflash_block_isbad;
+	mtd->_block_markbad = sflash_block_markbad;	
 	mtd->flags = (MTD_WRITEABLE | MTD_BIT_WRITEABLE); /* just like MTD_CAP_NORFLASH */
 	mtd->name = map->name;
 	mtd->writesize = 1;
diff --git a/drivers/mtd/maps/Makefile b/drivers/mtd/maps/Makefile
index 68a9a91..6d4c957 100644
--- a/drivers/mtd/maps/Makefile
+++ b/drivers/mtd/maps/Makefile
@@ -57,3 +57,8 @@ obj-$(CONFIG_MTD_VMU)		+= vmu-flash.o
 obj-$(CONFIG_MTD_GPIO_ADDR)	+= gpio-addr-flash.o
 obj-$(CONFIG_MTD_LATCH_ADDR)	+= latch-addr-flash.o
 obj-$(CONFIG_MTD_LANTIQ)	+= lantiq-flash.o
+
+ifeq ($(CONFIG_ARCH_ARMADA_XP),y)
+	include $(srctree)/arch/arm/mach-armadaxp/config/mvRules.mk
+	obj-$(CONFIG_MV_INCLUDE_SPI) += ../../../arch/arm/mach-armadaxp/flashmap.o
+endif
-- 
1.7.0

